src/Appwrite/Migration/Version/V19.php

```diff
@@ -0,0 +1,180 @@
+                $this->alterPermissionIndex($id);
+            }
+
+            usleep(50000);
+        }
+    }
+
+    /**
+     * Fix run on each document
+     *
+     * @param Document $document
+     * @return Document
+     */
+    protected function fixDocument(Document $document): Document
+    {
+        switch ($document->getCollection()) {
+            case 'projects':
+                /**
+                 * Bump version number.
+                 */
+                $document->setAttribute('version', '1.3.0');
+
+                /**
+                 * Set default passwordHistory
+                 */
+                $document->setAttribute('auths', array_merge([
+                    'passwordHistory' => 0,
+                    'passwordDictionary' => false,
+                ], $document->getAttribute('auths', [])));
+                break;
+            case 'users':
+                /**
+                 * Default Password history
+                 */
+                $document->setAttribute('passwordHistory', $document->getAttribute('passwordHistory', []));
+                break;
+            case 'teams':
+                /**
+                 * Default prefs
+                 */
+                $document->setAttribute('prefs', $document->getAttribute('prefs', new \stdClass()));
+                break;
+            case 'attributes':
+                /**
+                 * Default options
+                 */
+                $document->setAttribute('options', $document->getAttribute('options', new \stdClass()));
+                break;
+            case 'buckets':
+                /**
+                 * Set the bucket permission in the metadata table
+                 */
+                try {
+                    $internalBucketId = "bucket_{$this->project->getInternalId()}";
+                    $permissions = $document->getPermissions();
+                    $fileSecurity = $document->getAttribute('fileSecurity', false);
+                    $this->projectDB->updateCollection($internalBucketId, $permissions, $fileSecurity);
+                } catch (\Throwable $th) {
+                    Console::warning($th->getMessage());
+                }
+                break;
+        }
+
+        return $document;
+    }
+
+    protected function alterPermissionIndex($collectionName): void
+    {
+        $collectionName = "`{$this->projectDB->getDefaultDatabase()}`.`_{$this->project->getInternalId()}_{$collectionName}_perms`";
+
+        try {
+            $this->pdo->prepare("ALTER TABLE {$collectionName} DROP INDEX `_permission`")->execute();
+        } catch (\Throwable $th) {
+            Console::warning($th->getMessage());
+        }
+
+        try {
+            $this->pdo->prepare("ALTER TABLE {$collectionName} ADD INDEX `_permission` (`_permission`, `_type`, `_document`)")->execute();
+        } catch (\Throwable $th) {
+            Console::warning($th->getMessage());
+        }
```

Comment:
Let's do this in a transaction in a single try/catch

Reply:
Done!